src/libraries/System.Diagnostics.Process/tests/ProcessWaitingTests.cs

```diff
@@ -62,6 +119,21 @@ public void CurrentProcess_WaitNeverCompletes(int milliseconds)
             Assert.False(Process.GetCurrentProcess().WaitForExit(milliseconds));
         }
 
+        [Theory]
+        [InlineData(0)]  // poll
+        [InlineData(10)] // real timeout
+        public async Task CurrentProcess_WaitAsyncNeverCompletes(int milliseconds)
+        {
+            using (var cts = new CancellationTokenSource(milliseconds))
+            {
+                var token = cts.Token;
+                var process = Process.GetCurrentProcess();
+                OperationCanceledException ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => process.WaitForExitAsync(token));
```

Comment:
You're passing in a token that's already been canceled by the time you call this.  As such, WaitForExitAsync should actually complete synchronously, and it'd be better to lock that behavior into the test, e.g.
```C#
Task t = process.WaitForExitAsync(token);
Assert.Equal(TaskStatus.Canceled, t.Status);
OperationCanceledException ex = await Assert.ThrowsAnyAsync<OperationCanceledException>(() => t);
```